### PR TITLE
HPCC-15220 ConfigMgr - Add custom binding parameters

### PR DESCRIPTION
**Pull request by garonsky**
Signed-off-by: Gleb Aronsky gleb.aronsky@lexisnexis.com

### CODE REVIEW

**Comment by garonsky** [comment]
@rpastrana please review

**Comment by rpastrana** [comment]
@garonsky sent comment offline regarding XML structure

**Comment by garonsky** [comment]
@timothyklemm please confirm the configuration contains the data you need.

**Comment by stuartort** [comment]
@timothyklemm please review

**Comment by rpastrana** [comment]
Timm's comments were off-line:
I’m OK with free-form name value pairs. I’m confused by the element structure.

What does it mean for EspBinding to contain an element named EspBinding? Shouldn’t the inner EspBinding be CustomBindingParameters?

The use of a plural name for a singular entity is also confusing. I understand it is the default label for a list of entities within configmgr. When looking at the finished configuration, however, it suggests that each instance can represent multiple values.

Scapps developers frequently edit the esp.xml file directly when developing and testing our features. I even manually edited the esp.xml file used by my debug esp build when working with the HPCC platform. While neither issue is significant programmatically, I don’t think human readability should be discounted. This is what I expected:

<EspBinding …>
    <CustomBindingParameters>
        <CustomBindingParameter>
         …
    </CustomBindingParameters>
    …
</EspBinding>
@garonsky FYI

**Comment by garonsky** [comment]
@rpastrana please review

File: esp/services/WsDeploy/WsDeployService.cpp [hunk 1]
```diff
@@ -5305,6 +5305,19 @@ bool CWsDeployFileInfo::handleRows(IEspContext &context, IEspHandleRowsRequest &
         const char* compType = pRow->queryProp("@compType");
         xpath.clear().appendf("Override[@component='%s'][@instance='%s']", compType, rowName);
       }
+      else if (!strcmp(rowType, "CustomBindingParameter"))
```
Comment (rpastrana) [inline esp/services/WsDeploy/WsDeployService.cpp]: is it worthwhile to do a case insensitive compare? Or do we control the CustomBindingParameter label?

**Review by rpastrana — APPROVED** [review]
@garonsky minor comment

File: esp/services/WsDeploy/WsDeployService.cpp (hunk 1 above)
Comment (garonsky) [inline esp/services/WsDeploy/WsDeployService.cpp]: We contol this label, it is defined in the xsd.

**Comment by rpastrana** [comment]
@garonsky changes look reasonable, sample out you provided looked reasonable as well.

**Comment by garonsky** [comment]
@richardkchapman this is ready to merge.

**Pull request merged**